dotcom-rendering/src/components/FormatBoundary.tsx

```diff
@@ -0,0 +1,39 @@
+import { css } from '@emotion/react';
+import type { ArticleFormat } from '@guardian/libs';
+import { paletteDeclarations } from '../palette';
+import { useConfig } from './ConfigContext';
+
+type Props = {
+	format: ArticleFormat;
+	children: JSX.Element;
+};
+
+/**
+ * Used to create a set of palette colours for a different format
+ * to that of the :root element
+ */
+export const FormatBoundary = ({ format, children }: Props) => {
+	const { darkModeAvailable } = useConfig();
+
+	return (
+		<div
+			data-format-theme={format.theme}
+			data-format-design={format.design}
+			data-format-display={format.display}
+			css={[
+				css`
+					display: contents;
+
+					${css(paletteDeclarations(format, 'light'))}
+					@media (prefers-color-scheme: dark) {
+						${darkModeAvailable
+							? css(paletteDeclarations(format, 'dark'))
+							: '/* dark mode unavailable */'}
+					}
+				`,
+			]}
```

Comment:
It does, but we would need to prove that this is an actual performance issue:
- any identical CSS will be deduplicated by Emotion automatically
- I could not see a way of enforcing a strong contract that would ensure only required custom properties are defined
- slight efficiency should not get in the way of solving imminent blocking issues

As mentioned in the linked issue, once the palette is fully defined, it this would adds [about 4.5kB of CSS per article variation](https://docs.google.com/spreadsheets/d/1FWKCKFidfLIpgrX7mGIoUiHEh3brOkcQLjAuU7qRW0Y/edit) per format variation. If/when this becomes an issue, we’ll address it!

Reply:
Also worth noting: we already create many copies of the entirety of the card’s CSS as many of the styles are function that take `format` as an argument. So we might this might actually improve rather than worsen the current situation.